packages/dd-trace/src/config.js

```diff
@@ -287,237 +177,45 @@ class Config {
-    const DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED = coalesce(
-      options.traceId128BitGenerationEnabled,
-      process.env.DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED,
-      true
-    )
-
-    const DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED = coalesce(
-      options.traceId128BitLoggingEnabled,
-      process.env.DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED,
-      false
-    )
 
-    let appsec = options.appsec != null ? options.appsec : options.experimental && options.experimental.appsec
+    options = this.options = {
+      ...options,
+      appsec: options.appsec != null ? options.appsec : options.experimental && options.experimental.appsec
+    }
 
-    if (typeof appsec === 'boolean') {
-      appsec = {
-        enabled: appsec
+    if (typeof options.appsec === 'boolean') {
+      options.appsec = {
+        enabled: options.appsec
       }
-    } else if (appsec == null) {
-      appsec = {}
+    } else if (options.appsec == null) {
+      options.appsec = {}
     }
 
-    const DD_APPSEC_ENABLED = coalesce(
-      appsec.enabled,
-      process.env.DD_APPSEC_ENABLED && isTrue(process.env.DD_APPSEC_ENABLED)
-    )
-    const DD_APPSEC_RULES = coalesce(
-      appsec.rules,
-      process.env.DD_APPSEC_RULES
-    )
-    const DD_APPSEC_TRACE_RATE_LIMIT = coalesce(
-      parseInt(appsec.rateLimit),
-      parseInt(process.env.DD_APPSEC_TRACE_RATE_LIMIT),
-      100
-    )
-    const DD_APPSEC_WAF_TIMEOUT = coalesce(
-      parseInt(appsec.wafTimeout),
-      parseInt(process.env.DD_APPSEC_WAF_TIMEOUT),
-      5e3 // µs
-    )
-    const DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP = coalesce(
-      appsec.obfuscatorKeyRegex,
-      process.env.DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP,
-      `(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?)key)|token|consumer_?(?:id|key|se\
-cret)|sign(?:ed|ature)|bearer|authorization`
-    )
-    const DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP = coalesce(
-      appsec.obfuscatorValueRegex,
-      process.env.DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP,
-      `(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|to\
-ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:\\s*=[^;]|"\\s*:\\s*"[^"]+")|bearer\
-\\s+[a-z0-9\\._\\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\\w=-]+\\.ey[I-L][\\w=-]+(?:\\.[\\w.+\\/=-]+)?\
-|[\\-]{5}BEGIN[a-z\\s]+PRIVATE\\sKEY[\\-]{5}[^\\-]+[\\-]{5}END[a-z\\s]+PRIVATE\\sKEY|ssh-rsa\\s*[a-z0-9\\/\\.+]{100,}`
-    )
-    const DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML = coalesce(
-      maybeFile(appsec.blockedTemplateHtml),
-      maybeFile(process.env.DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML)
-    )
-    const DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON = coalesce(
-      maybeFile(appsec.blockedTemplateJson),
-      maybeFile(process.env.DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON)
-    )
     const DD_APPSEC_GRAPHQL_BLOCKED_TEMPLATE_JSON = coalesce(
```

Comment:
I'm a bit confused as to why some config entries have been converted to the new system, and some have not. Like this one for example, should be identical to the ones right above no ? why isn't it moved to the new system ?

Reply:
just being lazy *wink*, only converted [the ones reported by telemetry](https://github.com/DataDog/dd-go/blob/prod/trace/apps/tracer-telemetry-intake/telemetry-payload/static/config_norm_rules.json). But yeah we want to refactor all the config entries sometime